scripts/eurostat/health_determinants/common/EuroStat.py

```diff
@@ -0,0 +1,359 @@
+                 "{pv6}{pv7}{pv8}{pv9}{pv10}\n"
+                 "statType: dcs:measuredValue\n"
+                 "measuredProperty: dcs:count\n")
+
+_TMCF_TEMPLATE = ("Node: E:eurostat_population_{import_name}->E0\n"
+                  "typeOf: dcs:StatVarObservation\n"
+                  "variableMeasured: C:eurostat_population_{import_name}->SV\n"
+                  "measurementMethod: C:eurostat_population_{import_name}->"
+                  "Measurement_Method\n"
+                  "observationAbout: C:eurostat_population_{import_name}->geo\n"
+                  "observationDate: C:eurostat_population_{import_name}->time\n"
+                  "scalingFactor: 100\n"
+                  "value: C:eurostat_population_{import_name}->observation\n")
+
+
+class EuroStat:
+    """
+    This Class has requried methods to generate Cleaned CSV,
+    MCF and TMCF Files.
+    """
+
+    def __init__(self, input_files: list, csv_file_path: str,
+                 mcf_file_path: str, tmcf_file_path: str,
+                 import_name: str) -> None:
+        self._input_files = input_files
+        self._cleaned_csv_file_path = csv_file_path
+        self._mcf_file_path = mcf_file_path
+        self._tmcf_file_path = tmcf_file_path
+        self._import_name = import_name
+
+    def _parse_file(self, file_name: str, df: pd.DataFrame,
+                    import_name: dict) -> pd.DataFrame:
+        split_columns = df.columns.values.tolist()[0]
+        df = _split_column(df, split_columns)
+
+        split_columns = split_columns.replace('isced97', 'isced11')\
+                .replace('geo\time','geo').replace('geo\\time','geo')\
+                .replace('time\geo','time').replace('time\\geo','time')\
+                .replace('quantile', 'quant_inc')
+
+        df.rename(columns={
+            'geo\time': 'geo',
+            'geo\\time': 'geo',
+            'time\geo': 'time',
+            'time\\geo': 'time',
+            'isced97': 'isced11',
+            'quantile': 'quant_inc'
+        },
+                  inplace=True)
+
+        if import_name == "alcohol_consumption":
+            split_columns = split_columns.replace('frequenc',
+                                                  'frequenc_alcohol')
+            df.rename(columns={
+                'frequenc': 'frequenc_alcohol',
+            }, inplace=True)
+        elif import_name == "tobacco_consumption":
+            split_columns = split_columns.replace('frequenc',
+                                                  'frequenc_tobacco').replace(
+                                                      'quantile', 'quant_inc')
+            df.rename(columns={
+                'frequenc': 'frequenc_tobacco',
+                'quantile': 'quant_inc'
+            },
+                      inplace=True)
+        elif import_name == "bmi":
+            if 'quant_inc' in df.columns.values.tolist():
+                df = df[(~(df['quant_inc'] == 'UNK'))]
```

Comment:
Can such import specific statements be moved to an overloaded function, say `prepare_data()` so each import specific class can do it's own changes. That way when a new data set is added, the base class doesn't need to be modified.